sycl/doc/extensions/proposed/sycl_ext_intel_fpga_pipes_properties.asciidoc

```diff
@@ -0,0 +1,264 @@
+= SYCL_INTEL_FPGA_data_flow_pipes_properties
```

Comment:
Adding some general comments here, so the conversation can be threaded.

* My question about breaking API compatibility in #5838 is clearly related to this PR also.

* It's not clear to me why these properties should be in a separate extension.  Why not add them to the existing "sycl_ext_intel_dataflow_pipes" extension?

* Regardless of which extension they are in, we should define the behavior when these properties are used in a kernel that is submitted to a device that is **not** an FPGA.  I think the two options are:
  - These properties are an "optional kernel feature" as defined in [section 5.7](https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:optional-kernel-features) of the core SYCL spec.  In that case, submitting a kernel that uses these properties to a non-FPGA device would throw a synchronous exception at the time you submit the kernel.
  - These properties are defined to be ignored when the kernel is submitted to a non-FPGA device.  This is similar to what we did for the device global properties.

* The `min_capacity` property does not seem specific to FPGA at all.  Wouldn't this be relevant even if we implemented pipe on other devices?

Reply:
Thanks @gmlueck for the quick review!

@GarveyJoe, what are your thoughts on these and the other questions?

Reply:
API compatibility: I answered this in an email but I'll add it here for completeness: I think the solution is to keep min_capacity as a template parameter rather than pull it into the properties list.  While this might not be how we would have written it if we were building this from scratch, this will allow us to avoid breaking existing code.  min_capacity is sufficiently different from the other properties to warrant being separated from them.  While the other properties merely control the interface of the pipes, min_capacity is a platform-agnostic property of the algorithm itself.  

Separate extension: I'm not opposed to a single, combined extension.

Behaviour on non-FPGA targets: I prefer the 2nd option.  

min_capacity: Greg is correct, min_capacity is different from the other properties.  It can't be safely dropped as it is an algorithmic requirement needed to prevent deadlock.  It must be respected on all targets.